### PR TITLE
Include raw payload

### PR DESCRIPTION
**Pull request by laurensfischer**
Hey @semlette ,

thanks for your work and your awesome library. It works like a charm.
We need to read NDEFText tags where the first byte as well as the language code was not written correctly. Therefore we need some mechanism to access the raw payload, which we implemented in this pull request. Because we don't think it should be provided by default, we added it as optional parameter to the startReading method. 
We hope you appreciate our work and approve it to be merged. If not, please tell us what we should improve.

Best regards,
Laurens

### CODE REVIEW

**Comment by avbk** [comment]
:+1: 

Awesome, I was just looking for this. I also have badly formatted NFC tags. Hope this gets merged soon :slightly_smiling_face:

**Comment by semlette** [comment]
This seems like a good change.
The removal of the first byte stems back to v1, when I didn't even know what it was for - only in v2 did I realise.

Would the `dart:typed_data` `Uint8List` be a better data type than `String`?

Is there a particular reason why this should be controlled by an argument instead of always being set?

**Comment by laurensfischer** [comment]
Thanks for the input, `Uint8List` is indeed better. 
We thought some people won't need the raw payload, but it's probably easier to always provide the raw payload in case someone needs it.

File: lib/src/api.dart [hunk 1]
```diff
@@ -360,70 +366,89 @@ class NDEFRecord {
   /// null.
   final String languageCode;
 
+  /// rawPayload contains the raw payload provided by the reader.
+  /// It will only be set when reading NDEF tags. Otherwise it will be null.
```
Comment (semlette) [inline lib/src/api.dart:370]: The `NDEFRecord` wouldn't exist if the tag wasn't NDEF formatted so this comment is pointless.

Comment (laurensfischer) [reply lib/src/api.dart:370]: What I meant was that the field is only set when reading a tag, but not when instantiating a new record with one of the constructors.

Comment (semlette) [reply lib/src/api.dart:370]: Oh yeah.
I totally forgot writing was a thing. I don't do much NFC these days.
My bad!

**Comment by laurensfischer** [comment]
Were you already able to test the changes?

**Comment by semlette** [comment]
Afraid I don't have time to test it.

Will happily merge and release if you confirm that does indeed work.

@avbk, what do you think of this?

**Comment by avbk** [comment]
I just tried the branch and it seems to work pretty well :slightly_smiling_face:

**Comment by semlette** [comment]
Included in `v2.0.5`.

**Pull request merged**